yesod-auth/Yesod/Auth/Email.hs

```diff
@@ -383,19 +384,22 @@ registerHelper :: YesodAuthEmail master
                -> HandlerT Auth (HandlerT master IO) TypedContent
 registerHelper allowUsername dest = do
     y <- lift getYesod
+    req <- getRequest
     midentifier <- lookupPostParam "email"
-    let eidentifier = case midentifier of
-            Nothing -> Left Msg.NoIdentifierProvided
-            Just x
-                | Just x' <- Text.Email.Validate.canonicalizeEmail (encodeUtf8 x) ->
-                    Right $ normalizeEmailAddress y $ decodeUtf8With lenientDecode x'
-                | allowUsername -> Right $ TS.strip x
-                | otherwise -> Left Msg.InvalidEmailAddress
-
-    case eidentifier of
+    csrfToken <- lookupPostParam "_token"
```

Comment:
The error message used below says the the CSRF token can come from the headers or in the post parameters, but afaict the code only checks the post parameters. You could use `checkCsrfHeaderOrParam` to fix this. Also this would obviate the need to paste the CSRF error message from below